### PR TITLE
Release Kong chart 2.14.0

### PR DESCRIPTION
**Pull request by rainest**
#### What this PR does / why we need it:

Bumps versions and releases 2.14.0.

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] PR is based off the current tip of the `main` branch.
- [x] Changes are documented under the "Unreleased" header in CHANGELOG.md
- [x] Commits follow the [Kong commit message guidelines](https://github.com/Kong/kong/blob/master/CONTRIBUTING.md#commit-message-format)

### CODE REVIEW

**Review by pmalek — APPROVED** [review]
:shipit:

File: charts/kong/values.yaml [hunk 1]
```diff
@@ -121,10 +121,10 @@ extraLabels: {}
 # Specify Kong's Docker image and repository details here
 image:
   repository: kong
-  tag: "3.0"
+  tag: "3.1"
```
Comment (czeslavo) [inline charts/kong/values.yaml:124]: Are we sure we want to bump the gateway straight away given we have no compatibility tests passed yet? https://github.com/Kong/kubernetes-ingress-controller/issues/3220

Comment (rainest) [reply charts/kong/values.yaml:124]: Wait, we didn't do that before release?

Comment (rainest) [reply charts/kong/values.yaml:124]: Well, whatever, let https://github.com/Kong/kubernetes-ingress-controller/pull/3288#issuecomment-1358290461 run and if it passes we're good to go here.

**Pull request merged**